src/main/java/org/dcsa/ebl/service/impl/TransportDocumentTOServiceImpl.java

```diff
@@ -1,28 +1,207 @@
+                                                                } else {
+//                                                                    return Flux.concat(
+                                                                            return bookingService.findById(carrierBookingReference)
+                                                                                    .flatMap(booking -> {
+                                                                                        transportDocumentTO.setServiceTypeAtOrigin(booking.getServiceTypeAtOrigin());
+                                                                                        transportDocumentTO.setServiceTypeAtDestination(booking.getServiceTypeAtDestination());
+                                                                                        transportDocumentTO.setShipmentTermAtOrigin(booking.getShipmentTermAtOrigin());
+                                                                                        transportDocumentTO.setShipmentTermAtDestination(booking.getShipmentTermAtDestination());
+                                                                                        transportDocumentTO.setServiceContract(booking.getServiceContract());
+                                                                                        return Mono.just(booking);
+                                                                                    })
+//                                                                            ,shipmentService.findByCarrierBookingReference(carrierBookingReference)
+//                                                                                    .next()
+//                                                                                    .flatMap(shipment -> {
+//                                                                                        shipmentTransportService.findByShipmentIDOrderBySequenceNumber(shipment.getId())
+//                                                                                                .next()
+//                                                                                                .flatMap(shipmentTransport -> xxx)
+//                                                                                    })
+//                                                                    )
+//                                                                    .count()
+                                                                    .thenReturn(shippingInstruction);
+                                                                }
+                                                            }
+                                                    )
+                                                    .then(Mono.just(transportDocumentTO)),
```

Comment:
Is this needed ?  (If it is inside a flux just for the sake of side-effects, then you can often just remove it (or use `.then()`)